### DISCUSSION TITLE
What is the difference between `args.active` and `args.root` for ChkTeX?

### DISCUSSION THREAD

**Discussion by yarakos95** [James-Yu/LaTeX-Workshop#3305] [Q&A] [answered]
I do not understand these differences well.

The Wiki describes it as follows.

- `latex-workshop.chktex.args.active`

    > Linter arguments to check LaTeX syntax of the current file state in real time with ChkTeX.

- `latex-workshop.chktex.args.root`

    > Linter arguments to check LaTeX syntax of the entire project from the root file with ChkTeX.

For me these explanations are not enough :cry:

In practice, I feel that only `args.active` is sufficient for me. I do not know which files `args.root` applies to. Have I just not created a project with a directory structure where `args.root` would apply?

In addition, the lint content would not be changed within the same project. The intention of having two methods is also not clear.

## What I want you to tell me

I would like to know the specific directory structure in which each of these configurations works.

**Comment by jlelong** [comment, marked as answer]
The difference between the two configuration variables is not needed anymore. I just forgot to update the wiki. This is now done. Thanks pointing out this inconsistency.

With LW `8.26.0`, you should be using `latex-workshop.chktex.args.active`. In next releases, it will be replaced by `latex-workshop.chktex.args` simply.

**Reply by yarakos95** [reply]
Thank you for your response. I am pleased that my question has led to an improvement in LW!! :blush:

**Comment by yarakos95** [comment]
No one may be interested in this topic, but I leave it here as a record.

I did a bit of digging in the Git logs because I wanted to know how the ChkTeX arguments were split in two. It seems to be related to these PR.

- https://github.com/James-Yu/LaTeX-Workshop/pull/56
- https://github.com/James-Yu/LaTeX-Workshop/pull/57

It was not clear when `args.root` became unusable... :confounded:
